### PR TITLE
Lesson2

### PR DESCRIPTION
**Pull request by VeronikaVasilchikova**
**Date: 2020/02/28**

Here is my second home task.
The following features were imlemented here:

- a popup on “Profile” button click;
- add the following rules to validate form elements:
 a) “title“ must be filled in
 b) “year“ should be between 1970 and current 
 c) “votes“ must be less than 100000
 d) “rating“ cannot be empty or 0
- possiility to add new films to the datatable via the form by pressing the button “Add” in case of successefull validation of the form;
- in case of failed validation show the warning text using the invalidMessage of the related control;
- clear the form and validation marks by pressing button "Clear"

I will be glad to receive a feedback and your comments on this task.

### CODE REVIEW

**Comment by ByTiger** [comment]
I don't understand the reason, why right panel with form is jumping when validation failed. Why form can't be fixed size? (not friendly for users)
Format notes:
- mix between single and double quotes

File: index.html [hunk 1]
```diff
@@ -86,42 +107,24 @@
...
-              },
-              {
-                view:"text",
-                label:"Votes",
-                name:"votes"
+            elements:form_edit_films_data,
+            rules:{
+              title: webix.rules.isNotEmpty,
+              year:function(value) {
+                const today = new Date();
```
Comment (Zwillinge) [inline index.html]: you have already declared these vars (today, currentYear) together with form_edit_films_data.

File: index.html [hunk 2]
```diff
@@ -132,6 +135,45 @@
...
+              webix.message("Validation is failed! You should fix wrong data");
+            }
+          });
+
+          $$("btnClear").attachEvent("onItemClick", function() {
+            webix.confirm({
+              title: "Clear this form",
+              text: "Are you sure you want to clear this form?",
+              callback: function(result) {
+                if (result) {
```
Comment (Zwillinge) [inline index.html]: don't mix callback and promise here.  "ok" button actions should be placed into a resolved promise handler (together with confirmed message).

did you find `callback` in the docs? please point me to the article, it's legacy  syntax and should be cleared from docs.

Comment (VeronikaVasilchikova) [reply index.html]: Yes, I found callbck in docs. It's difficult to find it now but when I can I'll send this link

File: index.html (hunk 1 above)
Comment (VeronikaVasilchikova) [reply index.html]: I have already fixed this mistake

File: index.html (hunk 2 above)
Comment (VeronikaVasilchikova) [reply index.html]: I have already fixed this mistake. Could you check it again?

**Pull request merged**